### DISCUSSION TITLE
Questions on anonymous uploads

### DISCUSSION THREAD

**Discussion by santiyounger** [gavvvr/obsidian-imgur-plugin#4] [Q&A]
Hey thanks a lot for this awesome plugin!

I have a couple of questions and some things I'd love to know your opinion

I'd love to know more about anonymous upload in imgur. I read in the README that they are not connected or saved to our imgur account.

Is there however any way to access  all of my previous screenshots that I've pasted into Obsidian?

I know this must be more on Imgurs API limitation, but is there any way we could implement a way in which these pasted screenshots go to our saved imgur images?

──────────────────────────────────────────

Have you had any issues with these anonymous pasted images? In terms of have you seen the URL break / the pasted image disappear.

---

I love the convenience of this, as well as how much space we can free, compared to attaching every screenshot in the Obsidian attachments folder.

I think it's a great idea, I'd love to know your opinions on how good of a long term solution this is for being able to access screenshots.

perhaps more than anything, I'd like to know how reliable you feel Imgur anonymous uploads are.

Thanks a lot again, excellent work

**Comment by gavvvr** [comment]
Hi @santiyounger. First of all, thank you for your words, I am happy to hear that this little piece of work is found valuable for Obsidian users. You can also hit the star-button on the repo to show it's helpfulness

> Is there however any way to access all of my previous screenshots that I've pasted into Obsidian?

No, it is not possible with "anonymous upload" approach used right now. Below it the quote from [the docs](https://apidocs.imgur.com/#authorization-and-oauth):

> For public read-only and anonymous resources, such as getting image info, looking up user comments, etc. all you need to do is send an authorization header with your **client_id** in your requests. This also works if you'd like to upload images anonymously (**without the image being tied to an account**), or if you'd like to create an anonymous album. This lets us know which application is accessing the API.

---

> I know this must be more on Imgurs API limitation, but is there any way we could implement a way in which these pasted screenshots go to our saved imgur images?

I am sure this is possible and Imgur API easily allows it, but it would require much more effort from me. I will try to explain: to perform non-anonymous actions you need to obtain OAuth-related token. This either requires a callback-url on the app's side (which I am not sure if it is possible to register from plugins API) or some other way to obtain OAuth token without callback URL. Unfortunately, I did not have experience with OAuth, so initially I made it work the simplest way (with Anonymous upload approach requiring only client_id you specify manually).

So yes, I think it is possible and I also think it is a good idea to allow user to have specific album on imgur.com for note's images. But it is not implemented like this right now.

---

> Have you had any issues with these anonymous pasted images? In terms of have you seen the URL break / the pasted image disappear.

Do you mean if it happened to me that the pasted Image ever disappeared? No, never. I used hackmd before which also pastes images anonymously with client_id approach under the hood. I had notes which I did not open for ~ 1 year and images were still present.

I also used imgur long time ago to paste images on forums and did not remember they ever disappeared. 

> I'd like to know how reliable you feel Imgur anonymous uploads are.

I tried to find some guarantees about imgur image's TTL (for how long they are preserved since last accessed), and found the following stackexchange answer referring to imgur docs and saying that images are kept **forever**: https://webapps.stackexchange.com/a/75994/124389 

The link in the answer is invalid though and I did not find the same guarantee explicitly on today's current docs.
But I am still sure using imgur is a very reliable approach (even for your private notes). Looking at my previous experience with HackMD, If the image gets accessed (even once a couple of years), it will not disappear. You may ask: What if I do not open my image for a decade? Hard to say if it will be preserved then. But If you did not open your note for a decade, in my opinion there is a hard change you will ever need it. Usually I paste images from video lectures to my quick studying notes. Later they will be transformed to a reliable text, if it is a really useful piece of information. I like that Obsidian supports Mathjax, so I can write formulas in LaTeX syntax.

One more observation which can be related to ubiquity/reliability of imgur: when you post on stackoverflow/stackexchange, the image you paste will get uploaded to imgur.com too and gets embedded to markdown of your question/answer

Hope the above info will help you to understand how reliable the current approach is

**Reply by santiyounger** [reply]
Thanks a lot for the insightful and thoughtful reply. I'm happy to have asked since you answered a lot of my questions!

Sound good, it's good to see the way in which these anonymous uploads work. I'd agree that it sounds like a good way to quickly deal with screenshots. 

---

I still see the value on connecting to one's uploads to a personal imgur account, since it gives it a bit of a backup and also quick visualization of which screenshots have been saved.

It's good to know that'd this be possible, however, I totally understand if this is a feature that would require too much additional work, hopefully it's something that could be implemented in the future as more people discover the value of this awesome plugin.

Let me know if I should make a feature request for it, even if it could be a long-term goal, it might be good to see if others find value in this as well

Thanks again!

I just starred this repo, excellent work!

**Reply by gavvvr** [reply]
> I still see the value on connecting to one's uploads to a personal imgur account

Me too. It can give you a feeling of controlling your data:
- you can see all of them at once (making sure they will not disappear because of being stale), 
- or you can remove something you pasted by accidence

So the feature request is very welcomed👍🏻. As you said, other people can find it valuable, and among them there can be a person who has enough expertise to contribute a PR

**Reply by santiyounger** [reply]
awesome, just posted, it mostly resurfacing the great points you raised in here! Thanks!
